apis/v1alpha2/referencepolicy_types.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:categories=gateway-api,shortName=refpol
+// +kubebuilder:storageversion
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
```

Comment:
`namespace` of the policy itself might be helpful here.

Reply:
I think that adding extra columns is a good idea, I can see a case for both the `from` namespace and the `from` GroupKind  and `To` GroupKind or something. That will need joining support for print columns though, which I don't think is ready yet.

I think we should leave this for now and come back and add once we've used this for a little bit, and have a better idea what's useful on the command-line.

Reply:
Agree that this can be a later extension, one potential here would be to just include the following columns:

- from namespace
- from kind
- to kind

Although that would be ideal, we still don't have anyway to do this for a list of structs (kubectl would just output the first matching value for a jsonpath). I've added a note about this in https://github.com/kubernetes-sigs/gateway-api/issues/588.